src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexFindOptimizations.cs

```diff
@@ -742,6 +789,12 @@ internal enum FindNextStartingPositionMode
         /// <summary>A multi-character case-sensitive string at a fixed distance from the start of the pattern.</summary>
         FixedDistanceString_LeftToRight,
 
+        // The following modes are not vectorized.
```

Comment:
This is not true.

Reply:
I should have said "not _fully_ vectorized". I see some `IndexOf(Any)?` calls on these modes, but they still loop over each character and call `RegexCharClass.CharInClass` which can be pretty heavy if the character is not ASCII or not cached.

Is my understanding correct that we should prefer `LeadingMultiString` over modes that call `CharInClass`?

Reply:
I was refering to https://github.com/dotnet/runtime/blob/d344e72a84b98d84b3e34d37e9c59cf04930cf5b/src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexFindOptimizations.cs#L601-L604 but upon closer look this is technically an $O(1)$ operation since we call `CharInClass` once per set. In any case I fixed the vectorized mode detection logic.